xonsh/parsers/base.py

```diff
@@ -19,13 +22,55 @@
 from xonsh.tokenize import SearchPath, StringPrefix
 from xonsh.lazyasd import LazyObject
 from xonsh.parsers.context_check import check_contexts
+from xonsh.events import events
+from xonsh.tools import print_exception
```

Comment:
The parser seemed like the best place to put `on_transform_command` so that it's handled consistently across ptk, readline, jupyter, and scripts.